libvast/vast/file.hpp

```diff
@@ -0,0 +1,111 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#pragma once
+
+#include "vast/config.hpp"
+#include "vast/path.hpp"
+
+#include <caf/error.hpp>
+#include <caf/expected.hpp>
+
+#include <cstdint>
+
+namespace vast {
+
+/// A simple file abstraction.
+class file {
+  file(const file&) = delete;
+  file& operator=(const file&) = delete;
+
+public:
+/// The native type of a file.
+#ifdef VAST_POSIX
+  typedef int native_type;
+#else
+  typedef void native_type;
+#endif
+
+  /// The mode in which to open a file.
+  enum open_mode {
+    invalid,
```

Comment:
```suggestion
  enum class open_mode {
```

Reply:
Let's not introduce changes in this PR that touch a lot of downstream code. This is a reasonable change in general (but then it would make more sense to hoist it into the top-level namespace and call it `file_open_mode`) but stretches the scope here a bit too much.